### PR TITLE
Adding code of conduct

### PR DESCRIPTION
**Pull request by hkaiser**
This is a suggestion for a code of conduct for the HPX project. Please comment.

### CODE REVIEW

File: .github/CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  * Posting (or threatening to post) other people's personally identifying
+    information ("doxing").
+  * Personal insults, especially those using racist or sexist terms.
+  * Unwelcome sexual attention.
+  * Advocating for, or encouraging, any of the above behavior.
+  * Repeated harassment of others. In general, if someone asks you to stop,
+    then stop.
+
+* **When we disagree, try to understand why.** Disagreements, both social and
+  technical, happen all the time and Django is no exception. It is important
```
Comment (aserio) [inline .github/CODE_OF_CONDUCT.md]: You missed one :p

Comment (hkaiser) [inline .github/CODE_OF_CONDUCT.md]: Thanks, good catch

**Comment by hkaiser** [comment]
One open question is where to put this file. We have the option to leave it inside the .github directory. Alternatively we can put it into the root directory as well.

**Comment by hkaiser** [comment]
> I'll leave this open for a few more years to give others the ability to comment.

As soon as we merge this, github will close the issue. Should we create a (permanent) ticket, instead?

**Comment by sithhell** [comment]
On 06/26/2017 01:56 PM, Hartmut Kaiser wrote:
>     I'll leave this open for a few more years to give others the ability
>     to comment.
> 
> As soon as we merge this, github will close the issue. Should we create
> a (permanent) ticket, instead?

Guess I was overly confused when I wrote the comment, I meant *days* not
years. Sorry about the confusion. I guess if we want to change the code,
we can always file new issues/PRs.

**Comment by hkaiser** [comment]
> One open question is where to put this file. We have the option to leave it inside the .github directory. Alternatively we can put it into the root directory as well.

I guess it's ok to leave the file in the .github subdir since we now link to the CoC from the main README.

**Pull request merged**